src/compiler/parser.ts

```diff
@@ -8392,7 +8392,10 @@ namespace Parser {
             phaseModifier = SyntaxKind.TypeKeyword;
             identifier = isIdentifier() ? parseIdentifier() : undefined;
         }
-        else if (identifier?.escapedText === "defer" && token() !== SyntaxKind.FromKeyword) {
+        else if (

```

Comment:
Why isn't this just the same as what's done for `type` above? You can't mix `type` and `defer`, so I would have expectd both to parse in the same fashion.

Reply:
Daniel pointed out there's a difference between them in other ways like `import type foo, { bar } from "..."` is not legal as it's ambiguous whether or not `bar` is type or not. So, all good.

Reply:
I agree, but that's not really a point against what you suggested for before, because that validation happens after we figured out what AST we are going to build (kind of like an early error rather than a syntax error).

Reply:
Well technically we could still unify parsing for them and report an "early error" while still building the proper AST.

(I'm happy to both keep the code as-is or to update it)